lec06-pop-models.Rmd

```diff
@@ -2,3 +2,360 @@
+2. Which of the fixed points are **stable**? (If the population is slightly different from the fixed point, will it go towards it or away from it?)
+3. How do the above depend on parameters of the model?
+
+---------
+
+We will see how to answer these questions quantitatively soon. But let's start by looking at them
+qualitatively. We will draw a **phase portrait** of this system, and the techniques we use will be applicable
+to almost any model you might come across. 
+
+Let's start answering these questions qualitatively for the exponential growth model. There's a lot of
+information in the differential equation we can qualitatively read off. If we plot $dN/dt$ vs $N$, 
+the places where $dN/dt = 0$ are the fixed points of our system. Let's do this for the exponential growth model. 
+
+```{r}
+N_seq <- seq(-20,20) # make a sequence of N values to plot dN/dt against
+dN_dt = N_seq  # assume r = 1
+
+qplot(N_seq,dN_dt)+
+  geom_line(aes(x=N_seq,y=0*N_seq)) # a line at zero for visual aid
+```
+We can also tell whether the fixed points are stable or unstable by looking at the sign of $dN/dt$ 
+on either side of the fixed point. For a one-dimensional system, our phase portrait is a line of $N$. 
+We mark the fixed point(s) with circles, and we draw an arrow to the left wherever $dN/dt$ is negative, 
+and an arrow to the right wherever $dN/dt$ is positive in the regions between fixed points. 
+
+<!-- diagram of phase portrait --> 
+
+This model is a bit boring - there's only one fixed point, and it's unstable. This makes sense - 
+we are talking about exponential growth, after all. There's no upper limit. 
+
+--------
+
+Of course in real life there must be an upper limit to growth, otherwise the surface of the earth would be
+covered in bacteria several feet thick in a matter of hours. Let's do a quick calculation for fun - how long
+would it take for a single bacterial cell undergoing exponential growth to take over the world?
+
+```{r}
+cell_volume = 0.6 #micrometres cubed - E. coli cell volume is ~0.6 to 0.7 micrometres^3
+growth_rate = 3 # let's say cells divide three times per hour - every 20 minutes (max growth rate of E. coli in good food conditions)
+
+N_t <- function(N0,r,t) { # r is growth rate - divisions per hour, t is time in hours, N0 is starting population
+  return(N0*2^(r*t-1)) # the -1 is because the first doubling is included with the factor of 2
+}
+
+radius_earth = 6000 # approximately 6000 km 
+SA_earth = 4*pi*radius_earth^2 # surface area of earth in km^2
+
+surface_covered <- function(N0,r,t) { # calculate the volume taken up by bacteria in m on the earth's surface
+  bac_volume <- N_t(N0,growth_rate,t)*cell_volume # in micrometres cubed
+  height_km = bac_volume*(1e-27)/SA_earth # assume the earth is flat compared to the height of our surface - volume is surface area x height of surface above earth. This is height in km
```

Comment:
Might be clearer to write that out as 1e-3^3/1e6^2 or similar instead of the final -27.